pyproject.toml

```diff
@@ -12,11 +12,21 @@ dynamic = ["version"]
 [project.optional-dependencies]
 
 gpunufft = ["gpuNUFFT>=0.9.0", "cupy-cuda12x"]
+
 torchkbnufft = ["torchkbnufft", "cupy-cuda12x"]
-cufinufft = ["cufinufft<2.3", "cupy-cuda12x"]
-finufft = ["finufft"]
+torchkbnufft-cpu = ["torchkbnufft", "cupy-cuda12x"]
+torchkbnufft-gpu = ["torchkbnufft", "cupy-cuda12x"]
```

Comment:
those "CI backends" could be put all together and with a comment above for explaining. 
you could also do something like `torchkbnufft-cpu = ["mri-nufft[torchkbnufft]"]` to avoid repeating the dependency